pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java

```diff
@@ -190,7 +194,7 @@ public void registerListener(Consumer<Notification> listener) {
                 });
 
                 return null;
-            }, executor);
+            }, listenerExecutor);
```

Comment:
Can you explains more about why this `listenerExecutor` is added?

Reply:
We need to use a single thread to notify listeners.